sql/expression/autoincrement.go

```diff
@@ -116,6 +118,11 @@ func (i *AutoIncrement) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 	}
 	i.autoIncVal = NewLiteral(nextVal, i.Type())
 
+	err = ctx.Session.SetSessionVariable(ctx, lastInsertIDSessionVar, val)
```

Comment:
Wrong place for this

This is handled in insert.go, updateLastInsertId

But probably what you want to do is change session.go, SetLastQueryInfo, to do this automatically so the two values always change at the same time